pkg/cmd/cli/cmd/tag.go

```diff
@@ -289,74 +293,93 @@ func (o TagOptions) RunTag() error {
+				target = &imageapi.ImageStream{
+					ObjectMeta: kapi.ObjectMeta{
+						Name: destName,
+					},
+				}
 			}
 
-			// try to create the target if it doesn't exist
-			target = &imageapi.ImageStream{
-				ObjectMeta: kapi.ObjectMeta{
-					Name: destName,
-				},
+			if target.Spec.Tags == nil {
+				target.Spec.Tags = make(map[string]imageapi.TagReference)
 			}
-		}
 
-		if target.Spec.Tags == nil {
-			target.Spec.Tags = make(map[string]imageapi.TagReference)
-		}
+			msg := ""
+			if o.deleteTag {
+				// The user wants to delete a spec tag.
+				if _, ok := target.Spec.Tags[destTag]; !ok {
+					return fmt.Errorf("destination tag %s/%s does not exist.\n", o.destNamespace[i], destNameAndTag)
+				}
+				delete(target.Spec.Tags, destTag)
+				msg = fmt.Sprintf("Deleted tag %s/%s.", o.destNamespace[i], destNameAndTag)
+			} else {
+				// The user wants to symlink a tag.
+				targetRef, ok := target.Spec.Tags[destTag]
+				if !ok {
+					targetRef = imageapi.TagReference{}
+				}
 
-		msg := ""
-		if !o.deleteTag {
-			// The user wants to symlink a tag.
-			targetRef, ok := target.Spec.Tags[destTag]
-			if !ok {
-				targetRef = imageapi.TagReference{}
-			}
+				targetRef.From = &kapi.ObjectReference{
+					Kind: o.sourceKind,
+				}
+				localRef := o.ref
+				switch o.sourceKind {
+				case "DockerImage":
+					targetRef.From.Name = localRef.String()
+				default:
+					targetRef.From.Name = localRef.NameString()
+					targetRef.From.Namespace = o.ref.Namespace
+				}
 
-			targetRef.From = &kapi.ObjectReference{
-				Kind: o.sourceKind,
-			}
-			localRef := o.ref
-			switch o.sourceKind {
-			case "DockerImage":
-				targetRef.From.Name = localRef.String()
-			default:
-				targetRef.From.Name = localRef.NameString()
-				targetRef.From.Namespace = o.ref.Namespace
+				sameNamespace := o.namespace == o.destNamespace[i]
+				target.Spec.Tags[destTag] = targetRef
+				if o.aliasTag {
+					if sameNamespace {
+						msg = fmt.Sprintf("Tag %s set up to track %s.", destNameAndTag, o.ref.Exact())
+					} else {
+						msg = fmt.Sprintf("Tag %s/%s set up to track %s.", o.destNamespace[i], destNameAndTag, o.ref.Exact())
+					}
+				} else {
+					if sameNamespace {
+						msg = fmt.Sprintf("Tag %s set to %s.", destNameAndTag, o.ref.Exact())
+					} else {
+						msg = fmt.Sprintf("Tag %s/%s set to %s.", o.destNamespace[i], destNameAndTag, o.ref.Exact())
+					}
+				}
 			}
 
-			target.Spec.Tags[destTag] = targetRef
-			msg = fmt.Sprintf("Tag %s set up to track tag %s/%s.", o.ref.Exact(), o.destNamespace[i], destNameAndTag)
-		} else {
-			// The user wants to delete a spec tag.
-			if _, ok := target.Spec.Tags[destTag]; !ok {
-				return fmt.Errorf("destination tag %s/%s does not exist.\n", o.destNamespace[i], destNameAndTag)
+			// Check the stream creation timestamp and make sure we will not
+			// create a new image stream while deleting.
+			if target.CreationTimestamp.IsZero() && !o.deleteTag {
+				_, err = isc.Create(target)
```

Comment:
Eh.

On Nov 3, 2015, at 1:09 AM, Jordan Liggitt <notifications@github.com> wrote:

In pkg/cmd/cli/cmd/tag.go
<https://github.com/openshift/origin/pull/5626#discussion_r43718779>:

>  			}
>
> -			target.Spec.Tags[destTag] = targetRef
> -			msg = fmt.Sprintf("Tag %s set up to track tag %s/%s.", o.ref.Exact(), o.destNamespace[i], destNameAndTag)
> -		} else {
> -			// The user wants to delete a spec tag.
> -			if _, ok := target.Spec.Tags[destTag]; !ok {
> -				return fmt.Errorf("destination tag %s/%s does not exist.\n", o.destNamespace[i], destNameAndTag)
> +			// Check the stream creation timestamp and make sure we will not
> +			// create a new image stream while deleting.
> +			if target.CreationTimestamp.IsZero() && !o.deleteTag {
> +				_, err = isc.Create(target)

not uid or resourceVersion?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5626/files#r43718779>.